File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {

        String[] arrOfStr = logLine.split(":");

        return arrOfStr.length > 1 ? arrOfStr[1].trim() : "";
    }

    public static String logLevel(String logLine) {
        String message = new String();
        if(logLine.contains("ERROR")){
            message = "error";
        } else if(logLine.contains("WARNING")){
            message = "warning";
        } else if(logLine.contains("INFO")){
            message = "info";
        } else{
            message = "";
        }
        return message;
    }

    public static String reformat(String logLine) {
        String level = logLevel(logLine);
        String msg = message(logLine);
        String rev = msg + ' ' + "("+level+")";
        return rev;
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        string[] arrayfromString = logLine.Split(":");
        return arrayfromString.Length > 1 ? arrayfromString[1].Trim(): "";
    }

    public static string LogLevel(string logLine)
    {
        string[] arrayfromString = logLine.Split("]:");
        return arrayfromString[0].Substring(1).ToLower().Trim();
    }

    public static string Reformat(string logLine)
    {
        string message = Message(logLine);
        string logLevel = LogLevel(logLine);
        return ($"{message} ({logLevel})");
    }
}
